src/univariate/continuous/triangular.jl

```diff
@@ -93,20 +93,34 @@ entropy(d::TriangularDist{T}) where {T<:Real} = one(T)/2 + log((d.b - d.a) / 2)
 
 function pdf(d::TriangularDist, x::Real)
     a, b, c = params(d)
-    return if x < c
-        2 * max(x - a, 0) / ((b - a) * (c - a))
+    T = typeof(x / a)
+    if x < a || b < x
+        return T(0)
+    elseif a <= x < c
+        return 2 * (x - a) / ((b - a) * (c - a))
+    elseif x == c
+        return T(2) / (b - a)
+    elseif c < x <= b
+        return 2 * (b - x) / ((b - a) * (b - c))
     else
-        2 * max(b - x, 0) / ((b - a) * (b - c))
+        return T(NaN)
     end
```

Comment:
This branch can't be reached, can it? We also would never want to return `NaN` - inside of the support we would like a finite value, and outside of it 0.
It would also error in cases such as `pdf(::TriangularDist{Int}, ::Rational{Int})`:
```julia
julia> Rational{Int}(NaN)
ERROR: InexactError: Int64(NaN)
```

Reply:
I believe it can with:
pdf(TriangularDist{Int}(0, 2, 1), NaN)

I added this in because of the check you pointed me to in test_nonfinite():
    for f in (cdf, ccdf, logcdf, logccdf)
        if distr isa NoncentralT
            # broken in StatsFuns/R
            @test_broken isnan(f(distr, NaN))
        else
            @test isnan(f(distr, NaN))
        end
    end

    Having this last branch for NaN allows me to pass this NaN test.